### DISCUSSION TITLE
Filtering traffic on a bridge

### DISCUSSION THREAD

**Discussion by Fregf** [FoobarOy/foomuuri#46] [Q&A] [answered]
I have a system where I have added the public network interface eth0 added to bridge br0. Then the vnet interfaces of all virtual machines running on libvirt+qemu are added to that same bridge, so that all VMs get direct access to the public network.

In shorewall I would add br0 to the net zone, and then I could create policies and rules for all traffic going from the net to the net zone. I could even go further, and assign br0:eth0 to the net zone, and br0:vnet+ to the vm zone, and then I could use these zones to create rules. How do I do something similar in Foomuuri?

I tried assigning br0 to the public zone, and then create rules in a `public-public` section, but this did not work. Then I tried to create this zonemap:
```
zonemap {
	oifname vnet* new_dzone vm
	iifname vnet* new_szone vm
}
```

but rules in the public-vm are not trigged though.

Then I tried

```
forward {
        accept log                      
}
```

But still nothing is logged when I connect from the public network to a vm.

What would be the right way to filter traffic on the bridge?
I guess I need something which generates rules like in this example; https://wiki.nftables.org/wiki-nftables/index.php/Bridge_filtering#Example:_Stateful_bridge_firewall

**Comment by kimheino** [comment, marked as answer]
> I have a system where I have added the public network interface eth0 added to bridge br0. Then the vnet interfaces of all virtual machines running on libvirt+qemu are added to that same bridge, so that all VMs get direct access to the public network.

Foomuuri does not support filtering traffic between bridge ports (vnet0 <-> vnet1, or vnet0 <-> eth0 in your case).

You have two choices:

1) Install Foomuuri to virtual machines and filter traffic there, not on host.
2) Take eth0 out from bridge and route traffic eth0 <-> br0 instead of bridging it. Then assign br0 to `vm` zone and use normal `public-vm` filtering.

More detailed answer:

Foomuuri creates `inet` family firewall. In nftables bridging requires separate `bridge` family firewall in addition to normal inet. It's similar to inet, but there are lot of limitations too. Above two options are much better solutions to this.

See: https://wiki.nftables.org/wiki-nftables/index.php/Bridge_filtering
